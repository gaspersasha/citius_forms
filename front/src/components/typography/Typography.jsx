import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import s from './typography.module.sass';

/**
 * Typography component
 * Allows to configure rendered text element
 */
const Typography = ({ children, type, color, align, className, ...rest }) => {
  // defining element type to render
  const Component = /h[1-6]/.test(type) ? type : 'p';

  // composing selectors from provided props
  const composedClassName = cn(
    s.typographyComponent,
    s[`txt-${color}`],
    s[`txt-${align}`],
    className
  );

  return (
    <Component className={composedClassName} {...rest}>
      {children}
    </Component>
  );
};

Typography.propTypes = {
  // whatever will be rendered insine the component
  children: PropTypes.node.isRequired,
  // css selector to apply to the root element
  className: PropTypes.string,
  // content color value
  color: PropTypes.string,
  // text alignment value
  align: PropTypes.oneOf(['left', 'center', 'right']),
  // rendered elementy type (tag name basically)
  type: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p']),
};

Typography.defaultProps = {
  type: 'p',
  color: 'dark',
  align: 'left',
  className: '',
};

export default Typography;
