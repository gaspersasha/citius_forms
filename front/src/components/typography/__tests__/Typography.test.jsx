import React from 'react';
import Typography from '../Typography';

describe('<Typography />', () => {
  let wrapper;

  it('rendered without errors', () => {
    wrapper = shallow(<Typography>Test</Typography>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with basic parameters', () => {
    wrapper = shallow(<Typography>Test</Typography>);

    expect(wrapper.find('p').text()).toBe('Test');
    expect(wrapper.find('p').prop('className')).toEqual(
      'typographyComponent txt-dark txt-left'
    );
  });

  it('renders provided element type', () => {
    wrapper = shallow(<Typography type="h1">Test</Typography>);
    expect(wrapper.find('h1').exists()).toBe(true);
  });

  it('passes provided className', () => {
    wrapper = shallow(<Typography className="test">Test</Typography>);

    expect(wrapper.find('p').prop('className')).toEqual(
      'typographyComponent txt-dark txt-left test'
    );
  });

  it('changes color & align selectors', () => {
    wrapper = shallow(
      <Typography align="right" color="black">
        Test
      </Typography>
    );

    expect(wrapper.find('p').prop('className')).toEqual(
      'typographyComponent txt-black txt-right'
    );
  });
});
