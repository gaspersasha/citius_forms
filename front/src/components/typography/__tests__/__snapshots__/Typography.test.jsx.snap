// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> rendered without errors 1`] = `
<p
  className="typographyComponent txt-dark txt-left"
>
  Test
</p>
`;
